src/App.tsx

```
@@ -1,27 +1,79 @@
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [visibleList, setVisibleList] = useState<VisibList>(VisibList.empty);
+  const [error, setError] = useState<string | null>(null);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  useEffect(() => {
+    if (visibleList === VisibList.empty) {
+      return;
+    }
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    if (visibleList === VisibList.all) {
+      getAll()
+        .then(setGoods)
+        .catch(() => setError('Failed to load data'));
+    }
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    if (visibleList === VisibList.five) {
+      get5First()
+        .then(setGoods)
+        .catch(() => setError('Failed to load data'));
+    }
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    if (visibleList === VisibList.red) {
+      getRedGoods()
+        .then(setGoods)
+        .catch(() => setError('Failed to load data'));
+    }
+  }, [visibleList]);
+
+  const handleSwitchAll = () => {
+    return setVisibleList(VisibList.all);
+  };
+
+  const handleSwitchFive = () => {
+    return setVisibleList(VisibList.five);
+  };
+
+  const handleSwitchRed = () => {
+    return setVisibleList(VisibList.red);
+  };
+
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+
+      <button type="button" data-cy="all-button" onClick={handleSwitchAll}>
+        Load all goods
+      </button>
+
+      <button
+        type="button"
+        data-cy="first-five-button"
+        onClick={handleSwitchFive}
+      >
+        Load 5 first goods
+      </button>
+
+      <button type="button" data-cy="red-button" onClick={handleSwitchRed}>
+        Load red goods
+      </button>
+
+      {error !== '' && <p role="alert">{error}</p>}
```

The error message rendering condition `{error !== '' && <p role="alert">{error}</p>}` is incorrect. If `error` is `null`, this will still render `<p role="alert">null</p>`. This violates checklist item #3: 'Surface errors to the UI state in a verifiable way (e.g., set an error state and conditionally render an error message) rather than failing silently.'

Consider changing the condition to `error !== null && <p role="alert">{error}</p>` or simply `error && <p role="alert">{error}</p>` to ensure the error message only appears when there is an actual error string.